app/src/main/java/com/sunasterisk/thooi/ui/post/detail/PostDetailsVM.kt

```diff
@@ -28,19 +36,115 @@ class PostDetailsVM(
 
     private val _requestedPostId = MutableLiveData<String>()
 
+    private val _selectedFixerId = MutableLiveData("")
+
+    private val exceptionHandler = CoroutineExceptionHandler { _, _ ->
+        _toastStringRes.postValue(R.string.error_unknown)
+    }
+
     val postDetail = _requestedPostId.asFlow()
         .flatMapLatest { postDetailRepo.getPostDetailById(it) }
-        .catch { _errorRes.postValue(R.string.error_unknown) }
+        .catch { _toastStringRes.postValue(R.string.error_unknown) }
         .asLiveData()
 
     val postDetailsAdapterItems = postDetail.asFlow()
-        .mapLatest { it.toPostDetailsAdapterItem() }
+        .combine(_selectedFixerId.asFlow()) { postDetail: PostDetail, selectedFixerId: String ->
+            postDetail.toPostDetailsAdapterItem(selectedFixerId)
+        }
+        .conflate()
         .asLiveData()
 
-    private val _errorRes = MutableLiveData<Event<@StringRes Int>>()
-    val errorRes: LiveData<Event<Int>> get() = _errorRes
+    private val _toastStringRes = MutableLiveData<Event<@StringRes Int>>()
+    val toastStringRes: LiveData<Event<Int>> get() = _toastStringRes
+
+    fun loadPost(postId: String) {
+        _requestedPostId.postValue(postId)
+    }
```

Comment:
ô nên lm theo cách này nhé 
```suggestion
    private val _requestedPostId = MutableLiveData<String>()
    val  requestedPostId = LiveData<String>
    get() = _requestedPostId 
    
```
Bên Fragment ô updae qua viewmodel .requestedPostId = XXXX

To add LiveData
Change the type of the data variables in ViewModel to LiveData or MutableLiveData.
MutableLiveData is a LiveData object whose value can be changed. MutableLiveData is a generic class, so you need to specify the type of data that it holds.
To change the value of the data held by the LiveData, use the setValue() method on the LiveData variable.
To encapsulate LiveData
The LiveData inside the ViewModel should be editable. Outside the ViewModel, the LiveData should be readable. This can be implemented using a Kotlin backing property.
A Kotlin backing property allows you to return something from a getter other than the exact object.
To encapsulate the LiveData, use private MutableLiveData inside the ViewModel and return a LiveData backing property outside the ViewModel.

Reply:
Ở đây t chủ đích k để lộ `_requestedPostId` bên ngoài view model. T có biết dùng `backing property` cho live data trong view model nhưng mà vẫn cảm ơn ô tư vấn chỗ này